### PR TITLE
Tag component

### PR DESCRIPTION
**Pull request by ollips**
Task: task-988153

Link: [preview](http://localhost:1111/)

Creating tag component and adding relevant documentation.

There are 3 types of tag - static, dismissible, and interactive, and 3 types of views - outlined, filled, and primary.
 
Design - [Figma](https://www.figma.com/design/GvIcCw0tWaJVDSWD4f1OIW/Fluent-2-web?node-id=133557-2279&t=oWRDkUAktt9miDFa-0)

## Testing

1. Visit [tag](http://localhost:1111/components/tag.html) page
2. Review code

## Contributor checklist

- [x] Did you update the description of this pull request with a review link and test steps?
- [x] Did you submit a changeset? Changesets are required for all code changes.
- [x] Does your pull request implement complex UI logic (js/ts)? Consider adding an integration test to test your user flow.
- [x] Does your pull request add a new page to the documentation site? Add your new page for automated accessibility testing in /integration/tests/accessibility.

### CODE REVIEW

File: css/src/components/tag.scss [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+$tag-color-primary: $primary-active;
+$tag-color-primary-hover: $primary-dark;
+$tag-background-primary: $primary-background;
+
+.tag {
+	display: inline-flex;
+	align-items: center;
+	justify-content: center;
+	padding: $tag-padding-block $tag-padding-inline;
+	border: $tag-border-width solid $tag-border-color;
```
Comment (homing1) [inline css/src/components/tag.scss:33]: It looks like the default in the Fluent Figma is borderless and they have an outline variant. Do we need to match? Also, the tags currently used in Q&A seem to be borderless too.

Comment (ollips) [reply css/src/components/tag.scss:45]: I want to keep outline as default to align with other components. For filled variants we normally use the matching class and I don't want to mess with this pattern.

**Comment by BenDMyers** [comment]
Works great in operating system forced colors mode 👍🏻 

Dismissable & interactive tags in Night Sky contrast theme

File: site/src/components/tag.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+| Primary  | `.tag.tag-primary` | <button type="button" class="tag tag-primary"><span>Tag</span><span class="tag-close"><span class="icon" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 448" class="fill-current-color"><path d="M269.254 224l137.373 137.373-45.254 45.254L224 269.254 86.627 406.627l-45.255-45.254L178.746 224 41.373 86.627l45.255-45.255L224 178.746 361.373 41.373l45.254 45.255L269.254 224z"/></svg></span></span></button> | <button type="button" class="tag tag-primary is-hovered"><span>Tag</span><span class="tag-close"><span class="icon" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 448" class="fill-current-color"><path d="M269.254 224l137.373 137.373-45.254 45.254L224 269.254 86.627 406.627l-45.255-45.254L178.746 224 41.373 86.627l45.255-45.255L224 178.746 361.373 41.373l45.254 45.255L269.254 224z"/></svg></span></span></button> |
+
+### Interactive tag
+
+In addition to dismissibility, interaction tags can have one other action that can provide more information or actions related to what the tag represents.
```
Comment (BenDMyers) [inline site/src/components/tag.md:88]: For my own understanding - will the popover-style interaction for interactive tags be only non-dismissal interaction style for tags? Could, for instance, there ever been an "interactive" tag that's just a `<button>` and not a popover summary?

File: css/src/components/tag.scss (hunk 1 above)
Comment (homing1) [reply css/src/components/tag.scss:45]: Should there at least be a borderless variant then?

Comment (ollips) [reply css/src/components/tag.scss:45]: `border-none` atomic should do that. But wondering where do you see the mismatch?

File: site/src/components/tag.md (hunk 2 above)
Comment (ollips) [reply site/src/components/tag.md:88]: I don't think so, check out fluent's figma  - https://www.figma.com/design/GvIcCw0tWaJVDSWD4f1OIW/Fluent-2-web?node-id=133557-2279&t=q0Q551AFJqataUFz-0

File: css/src/components/tag.scss (hunk 1 above)
Comment (homing1) [reply css/src/components/tag.scss:45]: In the fluent v2 Figma and the Fluent v2 site, it looks to me that borderless is their default and outline/filled is a variant

image

https://fluent2.microsoft.design/components/web/react/tag/usage

File: site/src/components/tag.md (hunk 2 above)
Comment (BenDMyers) [reply site/src/components/tag.md:88]: In that case, I might suggest a rewording here:

```suggestion
Interactive tags can be used to open a popover that provides more information or contextual actions related to the tag's subject. Interactive tags can also be dismissible.
```

File: site/src/components/tag.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+To do this, nest the [popover component](./popover.md) inside a tag.
```
Comment (BenDMyers) [inline site/src/components/tag.md:90]:
```suggestion
To make an interactive tag, nest the [popover component](./popover.md) inside a tag.
```

File: css/src/components/tag.scss [hunk 4]
```diff
@@ -0,0 +1,205 @@
+/* stylelint-disable selector-max-specificity, no-descending-specificity  */
+
+$tag-padding-block: 0.4em !default;
```
Comment (wibjorn) [inline css/src/components/tag.scss:3]: Not a blocker, but these two values seem like they might have the untended effect of producing award pixel amounts. True? 
16*.4 = 6.4
16*6.5 = 10.4

Would .375em and .675em work instead?

File: css/src/components/tag.scss [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+	&:hover,
+	&.is-hovered {
+		.tag-close {
+			color: $tag-color-primary-hover;
+		}
+	}
+
+	&.tag-interactive {
+		.tag-popover .tag-summary,
```
Comment (wibjorn) [inline css/src/components/tag.scss:154]: Wondering if we really need this one. I might be mis-remembering, but wasn't popover designed to work with various triggers.

**Review by wibjorn — CHANGES_REQUESTED** [review]

**Review by wibjorn — DISMISSED** [review]

File: site/src/components/tag.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+| State    | Class              | Example                                  |
+| -------- | ------------------ | ---------------------------------------- |
+| Outlined | `.tag`             | <span class="tag">Tag</span>             |
+| Filled   | `.tag.tag-filled`  | <span class="tag tag-filled">Tag</span>  |
+| Primary  | `.tag.tag-primary` | <span class="tag tag-primary">Tag</span> |
+
+### Dismissable tags
+
+The only possible action that can be taken is dismissal.
```
[the remark this answers is not in the document]
Comment (ollips) [reply site/src/components/tag.md:63]: How do you feel about that:

Dismissible tags are tags that can be removed by the user. These tags include a close icon alongside the tag's content and the entire tag is treated as a click/tap target for the purposes of dismissal.

[the remark this answers is not in the document]
Comment (BenDMyers) [reply site/src/components/tag.md:63]: I like it!

File: site/src/components/tag.md (hunk 3 above)
Comment (ollips) [reply site/src/components/tag.md]: Updated

File: site/src/components/tag.md (hunk 2 above)
Comment (ollips) [reply site/src/components/tag.md]: Updated

File: site/src/components/tag.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (ollips) [reply site/src/components/tag.md]: Updated

File: css/src/components/tag.scss (hunk 5 above)
Comment (ollips) [reply css/src/components/tag.scss:154]: not sure what you mean by that, can you explain please?

Comment (wibjorn) [reply css/src/components/tag.scss:154]: I'm just trying to understand why we need new classes. Shouldn't popover just work? popover's trigger is supposed to work with other components

**Review by wibjorn — APPROVED** [review]

File: css/src/components/tag.scss (hunk 5 above)
Comment (ollips) [reply css/src/components/tag.scss:154]: oh, I see. Well, the way tag is styled in fluent, it's hover classes etc, doesn't match our buttons (which I could potentially use, but still some adjustments would be needed) so I created these helpers to make them look as Fluent does it.

**Pull request merged**